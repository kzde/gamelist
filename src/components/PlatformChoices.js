import React from 'react';
import styled from 'styled-components';
import { themeColor, margins, colorPalettes, textSizes } from '../utils/styleUtils';

const Wrapper = styled.div`
  padding: ${margins.medium};
  background: ${colorPalettes.lightGrey}
`
const Label = styled.span`
  line-height: 37px;
  text-align: center;
  font-size: ${textSizes.normalText};
  &+span {
    margin-left: 12px;
  }
  color: ${props => props.selected ? `${themeColor.selected}` : `${themeColor.default}`}
`;

const PlatformChoices = (props) => {
  return (
    <Wrapper>
      {props.choices.map((choice, index) => {
        return (
          <Label
            key={`${choice}${index}`}
            onClick={() => props.handleClick(choice)}
            selected={props.selectedPlatForm === choice}
          >
            {choice}
          </Label>
        )
      })}
    </Wrapper>
  )
}

export default PlatformChoices;