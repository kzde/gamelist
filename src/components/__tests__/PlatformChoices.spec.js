import React from 'react';
import renderer from 'react-test-renderer';
import PlatformChoices from '../PlatformChoices';

const props = {
  choices: ['platform', 'another platform'],
  selectedPlatform: 'platform',
  handleClick: () => {}  
}

describe('PlatformChoices', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(<PlatformChoices {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
