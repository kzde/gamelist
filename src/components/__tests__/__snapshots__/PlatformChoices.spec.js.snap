// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformChoices should renders correctly 1`] = `
<div
  className="PlatformChoices__Wrapper-sc-1bsu13e-0 jyEzSo"
>
  <span
    className="PlatformChoices__Label-sc-1bsu13e-1 jeGRun"
    onClick={[Function]}
    selected={true}
  >
    platform
  </span>
  <span
    className="PlatformChoices__Label-sc-1bsu13e-1 iIEBRI"
    onClick={[Function]}
    selected={false}
  >
    another platform
  </span>
</div>
`;
